rpc/block_svc_routines.c

```diff
@@ -211,6 +212,36 @@ mapJsonFlagToJsonCstring(int jsonflag)
 }
 
 
+void collectRequestLogs(char *cmd, int errCode, const char *message, int json)
```

Comment:
Split printing the cmd and response printing.

See the comment I added yesterday above.
1. we need to print the cmd right after GB_METALOCK_OR_GOTO()
2. and grab the over all response from 'reply->out' right after block#OPT#CliFormatResponse() function calls

also you can use LOG("history", ...) for printing the log, just like we use LOG() for cli/mgmt/gfapi

Reply:
Why do we need to split printing cmd and response string?

Can't we print cmd string and 'reply->out string' together right after block#OPT#CliFormatResponse() function calls?

Reply:
> Why do we need to split printing cmd and response string?
> 
> Can't we print cmd string and 'reply->out string' together right after block#OPT#CliFormatResponse() function calls?

@bhumikagoyal 

Consider a case where daemon dies between acquiring the lock and receiving the remote responses. In this case we will not have any evidence of a cli request.

Hence it is better to split this and dump the command request details, right after the lock acquisition.